src/logging.rs

```diff
@@ -17,32 +17,40 @@ thread_local! {
 pub mod trace {
     extern crate tracing_log;
     use super::{clock, Context as LegacyContext, CONTEXT as LEGACY_CONTEXT};
-    use std::{env, error, fmt, time::Instant};
+
+    use std::{env, error, fmt, str, time::Instant};
     pub use tracing::*;
     pub use tracing_fmt::*;
 
     type SubscriberBuilder = Builder<default::NewRecorder, Format, filter::EnvFilter>;
     pub type Error = Box<error::Error + Send + Sync + 'static>;
 
+    #[derive(Clone)]
+    pub struct LevelHandle {
+        inner: filter::reload::Handle<filter::EnvFilter, default::NewRecorder>,
+    }
+
     /// Initialize tracing and logging with the value of the `ENV_LOG`
     /// environment variable as the verbosity-level filter.
-    pub fn init() -> Result<(), Error> {
+    pub fn init() -> Result<LevelHandle, Error> {
         let env = env::var(super::ENV_LOG).unwrap_or_default();
         init_with_filter(env)
     }
 
     /// Initialize tracing and logging with the provided verbosity-level filter.
-    pub fn init_with_filter<F: AsRef<str>>(filter: F) -> Result<(), Error> {
-        let dispatch = Dispatch::new(
-            subscriber_builder()
-                .with_filter(filter::EnvFilter::from(filter))
-                .finish(),
-        );
+    pub fn init_with_filter<F: AsRef<str>>(filter: F) -> Result<LevelHandle, Error> {
+        // Set up the subscriber
+        let builder = subscriber_builder()
+            .with_filter(filter::EnvFilter::from(filter))
+            .with_filter_reloading();
```

Comment:
I don't understand this change.  What is filter reloading and what effect does it have?

Reply:
The "filter" is what determines whether a span or event should be recorded. `with_filter_reloading` configures the builder to use a filter that can be dynamically changed after the subscriber is constructed. 

The method `reload_handle` that we [call on the subsequent line](https://github.com/linkerd/linkerd2-proxy/pull/279/files/553f4897cdeca55), which returns a handle that we can use to set a new filter, is only implemented when the filter on the subscriber being built has been wrapped to support reloading (which is what `with_filter_reloading` does).